### PR TITLE
Implemented db_setup file

### LINKED ISSUE #15154: Implement user creation and db creation in new installer
In issue https://github.com/HGustavs/LenaSYS/issues/14987 a plan for the early stages of a new installer was implemented. And in https://github.com/HGustavs/LenaSYS/issues/15019 we were tasked with creating new issues for implementing the missing functions from this plan.

Requirements:
- function to create new db.
- function to create new user.
- function to add user permissions for created db.

### PR DESCRIPTION
**Pull request by a22willi**
### Implemented db_setup
Pull request contains db_setup file with the following functions:
- Create database
- Create user
- Delete database
- Delete user
- Set permissions for user on database
- run sql file
- verbose mode for running sql file

### How to test?

Append the code below to the file `newinstaller/installer.php`. 
Change the pdo declaration to your mysql root credentials. 
This will modify your database, so ensure that you dont accidentally delete anything.
Also make sure that you try to change some of the settings to ensure that it all works properly. You could for example test the verbose mode for the function `execute_sql_file()`.

```php
try {
    $hostname = 'db';
    $username = 'root';
    $password = 'password';
    
    $dsn = "mysql:host=$hostname;charset=utf8mb4";
    $pdo = new PDO($dsn, $username, $password);
    $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
    echo "Connected successfully to MySQL server" . "<br>";
} catch (PDOException $e) {
    echo "Connection failed: " . $e->getMessage();
}

$installer = new DBSetup(pdo: $pdo, db_name: "LenaSYStest", db_user: "LenaTest", db_user_password: "test", hostname:"%");
echo $installer->create_db()['message'] . "<br>";
echo $installer->create_user(force: true)['message'] . "<br>";
echo $installer->set_permissions()['message'] . "<br>";

ob_start();
function callback($message) {
    echo $message . "<br>";
    ob_flush();
    flush();
}
echo $installer->execute_sql_file("../Shared/SQL/init_db.sql", "callback", verbose: false)["message"] . "<br>";

echo $installer->drop_user()['message'] . "<br>";
echo $installer->drop_db()['message'] . "<br>";
```
The output should look like this:
image

closes #15154

### CODE REVIEW

File: newinstaller/tools/db_setup.php [hunk 1]
```diff
@@ -1,61 +1,378 @@
...
+
+    /**
+     * function create_db
+     * Create db using class variables.
+     * Force first drops any existing database.
+     */
+    public function create_db(string $db_name = null, bool $force = false): array {
+        try {
+
+            $db_name = $db_name ?? $this->db_name;
```
Comment (SleepyBearIV) [inline newinstaller/tools/db_setup.php:29]: unnecessary spacing

File: newinstaller/tools/db_setup.php [hunk 2]
```diff
@@ -1,61 +1,378 @@
...
+                $this->sanitize($db_name);
+            }
+
+            $this->pdo->exec("CREATE DATABASE `$db_name`");
             $this->db_name = $db_name;
-            $this->db_user = $db_user;
+
+        } catch (PDOException $e) { 
+
+            $this->handle_exception($e, "Failed to create database {$db_name}. " .
```
Comment (SleepyBearIV) [inline newinstaller/tools/db_setup.php:42]: unnecessary spacing

File: newinstaller/tools/db_setup.php [hunk 3]
```diff
@@ -1,61 +1,378 @@
...
+
+    /**
+     * function create_user
+     * Creates a new sql user.
+     * Force first removes user.
+     */
+    public function create_user(string $user_name = null, string $password = null, string $hostname = null, bool $force = false): array {
+        try {
+
+            $user_name = $user_name ?? $this->db_user;
```
Comment (SleepyBearIV) [inline newinstaller/tools/db_setup.php:57]: unnecessary spacing

File: newinstaller/tools/db_setup.php [hunk 4]
```diff
@@ -1,61 +1,378 @@
...
+    }
+
+    /**
+     * function drop_db
+     * Drops an existing database. 
+     */
+    public function drop_db(string $db_name = null, bool $force = false): array { 
+        try {
+
+            $db_name = $db_name ?? $this->db_name;
```
Comment (SleepyBearIV) [inline newinstaller/tools/db_setup.php:99]: unnecessary spacing

File: newinstaller/tools/db_setup.php [hunk 5]
```diff
@@ -1,61 +1,378 @@
...
-        public function drop_db() { 
-            // TODO: implement function that drops db.
+    /**
+     * function drop_user
+     * removes an already existing user.
+     */
+    public function drop_user(string $user_name = null, string $hostname = null, bool $force = false): array {
+        try {
+
+            $user_name = $user_name ?? $this->db_user;
```
Comment (SleepyBearIV) [inline newinstaller/tools/db_setup.php:121]: unnecessary spacing

File: newinstaller/tools/db_setup.php [hunk 6]
```diff
@@ -1,61 +1,378 @@
...
-         */
-        public function drop_user(string $hostname = hostname) {
-            // TODO: Implement function that removes a user.
+            $this->pdo->exec("GRANT ALL PRIVILEGES ON `$db_name`.* TO `$user_name`@`$hostname`");
+
+            return $this->handle_success("Successfully set permissions for {$user_name}@{$hostname} on {$db_name}.");
+        } catch (PDOException $e) {
+
+            return $this->handle_exception($e, $e->getCode() == 42000 ?
+            "User does not exist. Failed to set permissions for {$user_name}@{$hostname} on {$db_name}." :
```
Comment (SleepyBearIV) [inline newinstaller/tools/db_setup.php:159]: unnecessary spacing

**Review by SleepyBearIV — CHANGES_REQUESTED** [review]
the output is the same on my pc, maybe it a good ide to test this on an xampp system as well because i know this coder uses docker and so do i, just a few unnecessary spacing fixes and it LGTM!

i also see that all indentations is made whit space and not tab, mabey this should be fixed as swell

**Review by SleepyBearIV — APPROVED** [review]
LGTM

**Comment by d22mikwe** [comment]
LGTM

**Pull request merged**